runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/CoreValidationsProvider.java

```diff
@@ -34,6 +36,8 @@ public class CoreValidationsProvider implements ValidationsProvider {
 
   private boolean ignoreParamsWithProperties;
 
+  private ArtifactAstDependencyGraphProvider artifactAstDependencyGraphProvider = new CachingArtifactAstGraphDependencyProvider();
```

Comment:
this would still be creating the dependencyGraph for each validation provider requiring it.